### PR TITLE
feat: integrate ticket statistics into ticket dashboard as a core feature

### LINKED ISSUE #4672: Make ticket statistics a core feature and integrate it into the ticket dashboard
Eventyay currently has a `Statistics` plugin that adds statistics as a dedicated page in the ticketing component.

Statistics should be a core platform feature, not a separate plugin. It should be available by default and integrated directly into the ticket dashboard.

Image
Image
Image

## Required changes

* Remove `Statistics` from the plugin view.
* Enable ticket statistics by default for all events with ticketing.
* Remove the dedicated ticket `Statistics` page.
* Integrate the existing ticket statistics into the ticket dashboard.
* Show the statistics section above `Internal comment`.

## Expected behavior

* Ticket statistics are always available without enabling a plugin.
* Organisers see ticket statistics directly on the ticket dashboard.
* The ticket dashboard shows statistics before the `Internal comment` section.
* There is no separate `Statistics` plugin entry.
* There is no separate ticket `Statistics` page in the sidebar.

## Acceptance criteria

* `Statistics` no longer appears as a plugin.
* Ticket statistics are available by default.
* Existing statistics charts/data are shown on the ticket dashboard.
* Statistics appear above `Internal comment`.
* The dedicated ticket statistics page and sidebar entry are removed or redirected.
* Existing statistics functionality continues to work.
* Permissions remain unchanged.

### PR DESCRIPTION
**Pull request by prayag78**
Fixes #4672

## Description
Integrates ticket statistics directly into the core ticket dashboard so statistics charts are available by default for all events without requiring a separate plugin to be enabled.

### Changes Made:
1. Removed plugin metadata from `apps.py` and default plugin configuration in `consts.py` so statistics is no longer toggled as an optional plugin.
2. Rendered Orders by Day, Revenue over Time, Orders by Product, and Seating statistics directly inside the ticket dashboard (`pretixcontrol/event/index.html`) above the Internal Comment section.
3. Statistics remain protected under the `can_view_orders` permission and are automatically hidden when an event has no orders.
4. Added a permanent redirect from `/control/event/<org>/<event>/statistics/` to the ticket dashboard in `control/urls.py` to preserve link compatibility.
5. Updated `clear_cache` signal handler in `signals.py` to properly purge all suffixed cache keys (`all` and subevent IDs) when orders are placed or paid.
6. Added test suite in `tests/tickets/control/test_statistics_dashboard.py` covering access permissions, HTTP 301 redirects, empty-state rendering, and cache invalidation.

Screenshot 2026-08-04 002323

https://github.com/user-attachments/assets/83e8c186-0983-4f9b-8f7f-c25d1fd88d25

## Summary by Sourcery

Integrate ticket statistics into the core event dashboard and retire the standalone statistics plugin view.

New Features:
- Display orders-by-day, revenue-over-time, orders-by-product, and seating statistics directly on the event dashboard for users with order-view permissions.

Bug Fixes:
- Ensure statistics cache keys for orders and subevents are fully invalidated when orders are placed or paid.

Enhancements:
- Expose statistics data via a reusable backend helper and wire it into the event dashboard context.
- Adjust content security policy for the event dashboard to allow required statistics scripting.
- Preserve compatibility of the legacy statistics URL by permanently redirecting it to the event dashboard.

Tests:
- Add dashboard integration tests covering statistics visibility by permission, legacy URL redirects, empty-order behavior, and cache invalidation.

### CODE REVIEW

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
Please address comments by copilot

File: app/eventyay/plugins/statistics/views.py [hunk 1]
```diff
@@ -2,260 +2,248 @@
...
 
-        subevent = None
-        if self.request.GET.get('subevent', '') != '' and self.request.event.has_subevents:
-            i = self.request.GET.get('subevent', '')
-            try:
-                subevent = self.request.event.subevents.get(pk=i)
-            except SubEvent.DoesNotExist:
-                pass
+    cache = request.event.cache
+    ckey = str(subevent.pk) if subevent else 'all'
```
[the remark this answers is not in the document]
Comment (prayag78) [reply app/eventyay/plugins/statistics/views.py:35]: fixed

**Review by Saksham-Sirohi — APPROVED** [review]

**Review by ArnavBallinCode — APPROVED** [review]
LGTM

Image
